cmd/mqtt_producer/main.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/AarC10/GSW-V2/lib/tlm"
+	"github.com/AarC10/GSW-V2/proc"
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+)
+
+var (
+	shmDir      = flag.String("shm", "/dev/shm", "directory to use for shared memory")
+	brokerURL   = flag.String("broker", "tcp://127.0.0.1:1883", "mqtt broker url")
+	topicPrefix = flag.String("topic_prefix", "gsw", "mqtt topic prefix")
+)
+
+func main() {
+	flag.Parse()
+
+	configData, err := proc.ReadTelemetryConfigFromShm(*shmDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_, err = proc.ParseConfigBytes(configData)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	opts := mqtt.NewClientOptions()
+	opts.AddBroker(*brokerURL)
+	opts.SetClientID("gsw-mqtt-app")
+	opts.SetCleanSession(true)
+
+	client := mqtt.NewClient(opts)
+	if token := client.Connect(); token.Wait() && token.Error() != nil {
+		log.Fatal(token.Error())
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	var wg sync.WaitGroup
+
+	for _, packet := range proc.GswConfig.TelemetryPackets {
+		wg.Add(1)
+		go func(packet tlm.TelemetryPacket) {
+			defer wg.Done()
+			err := packetWriter(ctx, packet, client)
+			if err != nil && !errors.Is(err, context.Canceled) {
+				log.Printf("error in writer: %v", err)
+			}
+		}(packet)
+	}
+
+	go func() {
+		sigChan := make(chan os.Signal, 1)
+		signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+		<-sigChan
+		log.Println("shutting down")
+		cancel()
+	}()
+
+	wg.Wait()
+	client.Disconnect(250)
+}
+
+func packetWriter(ctx context.Context, packet tlm.TelemetryPacket, client mqtt.Client) error {
```

Comment:
Another thing I would suggest is adding a way to configure downsampling in the future. If this is mainly for sending to visualization apps that read over MQTT we may not want to be publishing certain things at an insane rate that we can't perceive fast enough or to avoid nuking a computer trying to render millions of data points on a graph